src/components/course/tests/CoursePage.test.jsx

```diff
@@ -0,0 +1,166 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import { MemoryRouter } from 'react-router-dom';
+import { AppContext } from '@edx/frontend-platform/react';
+import { sendEnterpriseTrackEvent } from '@edx/frontend-enterprise-utils';
+import { CourseEnrollmentsContext } from '../../dashboard/main-content/course-enrollments/CourseEnrollmentsContextProvider';
+import { CourseContextProvider } from '../CourseContextProvider';
+import { UserSubsidyContext } from '../../enterprise-user-subsidy/UserSubsidy';
+import { SubsidyRequestsContext, SUBSIDY_TYPE } from '../../enterprise-subsidy-requests';
+import { initialCourseState } from '../../../utils/tests';
+import CoursePage from '../CoursePage';
+import { useAllCourseData } from '../data/hooks';
+
+jest.mock('@edx/frontend-enterprise-utils', () => ({
+  ...jest.requireActual('@edx/frontend-enterprise-utils'),
+  sendEnterpriseTrackEvent: jest.fn(),
+}));
+jest.mock('../data/hooks', () => ({
+  useAllCourseData: jest.fn(() => ({
+    isLoading: false,
+    courseData: {
```

Comment:
"lovely" (not) example of the test mock data issue you're hoping to solve. kudos for you to be proactive in trying to simplify mock test data for our future selves.